crates/c-api/include/wasmtime.h

```diff
@@ -101,6 +98,14 @@ WASM_API_EXTERN wasm_instance_t* wasmtime_linker_instantiate(
     own wasm_trap_t **trap
 );
 
+typedef struct wasmtime_caller_t wasmtime_caller_t;
+
+typedef own wasm_trap_t* (*wasmtime_func_callback_t)(const wasmtime_caller_t* caller, const wasm_val_t args[], wasm_val_t results[]);
+
+WASM_API_EXTERN own wasm_func_t* wasmtime_func_new(wasm_store_t*, const wasm_functype_t*, wasmtime_func_callback_t callback);
```

Comment:
Perhaps we could go ahead with the fully general `new_with_env` custom function for wasmtime itself?

Reply:
Can do.  I'll see about sharing the impls between these and the `wasm_` counterparts the best we can, too.